### PR TITLE
Vulnerability Fixes

### PR DESCRIPTION
**Pull request by TeaWithLucas**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SSCS-11066

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: app/client/javascript/request-type.ts [hunk 1]
```diff
@@ -8,10 +8,12 @@ export class RequestType {
   }
 
   requestOptionSelectEventListeners(): void {
-    const requestOptions = document.querySelector('#requestOptions');
+    const requestOptions: HTMLSelectElement =
+      document.querySelector('#requestOptions');
     if (requestOptions) {
       requestOptions.addEventListener('change', (input: any) => {
-        document.forms['request-option-form'].submit();
+        const forms: HTMLCollectionOf<HTMLFormElement> = document.forms;
```
[the remark this answers is not in the document]
Comment (TeaWithLucas) [reply app/client/javascript/request-type.ts:15]: Sure, ill change that

File: app/server/controllers/active-cases.ts [hunk 2]
```diff
@@ -2,11 +2,20 @@ import { Router, Request, Response } from 'express';
...
+import { getCasesByName } from '../utils/fieldValidation';
+import { CaseDetails } from '../services/cases';
+import { Dependencies } from '../routes';
 
 const logger = Logger.getLogger('active-cases.js');
 
-function getActiveCases(req: Request, res: Response) {
+function filterActiveCase(selectedHearing, index, array): boolean {
+  return !(
+    selectedHearing.appeal_details.state === 'dormantAppealState' ||
```
[the remark this answers is not in the document]
Comment (TeaWithLucas) [reply app/server/controllers/active-cases.ts:13]: didn't see that, was just recording the function as it is used by getActiveCases.
I'll fix that and move them to cases as there is a opposite function isCaseDormant I can just invert

File: app/server/controllers/login.ts [hunk 3]
```diff
@@ -131,23 +131,29 @@ function getIdamCallback(
...
 
-      const { statusCode, body }: rp.Response =
-        await hearingService.getOnlineHearingsForCitizen(
-          email,
-          req.session['tya'],
-          req
-        );
+      let statusCode: number = null;
+      let body: Array<CaseDetails> = null;
+      ({ statusCode, body } = await caseService.getCasesForCitizen(
```
[the remark this answers is not in the document]
Comment (TeaWithLucas) [reply app/server/controllers/login.ts:136]: I will fix this in a new ticket, I have noticed through seeing this that this is using request (and packages based on it) which are deprecated and a couple already causing problems, I will move to node-fetch in the new ticket

File: app/server/controllers/outcome.ts [hunk 4]
```diff
@@ -1,13 +1,21 @@
...
 import { Logger } from '@hmcts/nodejs-logging';
 import { TrackYourApealService } from '../services/tyaService';
 import { dateFormat } from '../utils/dateUtils';
+import { Dependencies } from '../routes';
 
 const logger = Logger.getLogger('outcome.js');
 
+function reformatOutcomeDates(outcomes): void {
+  outcomes.forEach((outcome) => {
+    const outcomeDate = dateFormat(outcome.date, 'DD-MM-YYYY');
```
[the remark this answers is not in the document]
Comment (TeaWithLucas) [reply app/server/controllers/outcome.ts:13]: I think I got what you meant, I have mapped it on the rendering side in outcome-tab.njk

**Review by pk2453 — APPROVED** [review]
Though some types can be provided but considering its in existing code and its already a big PR, so no changes considered in this regard. Rest looks good.

**Review by damien-cooke — APPROVED** [review]

File: app/server/controllers/active-cases.ts [hunk 5]
```diff
@@ -17,20 +26,13 @@ function getActiveCases(req: Request, res: Response) {
     AppInsights.trackEvent('MYA_SESSION_READ_FAIL');
   }
 
-  const hearings = session['hearings']!;
-  const activeCases = hearings.filter(filterActiveCase);
-  const activeHearingsByName = getHearingsByName(activeCases);
-  return res.render('active-tab.html', { activeHearingsByName });
+  const cases: Array<CaseDetails> = session['cases'] ? session['cases'] : [];
+  const activeCases = cases.filter(filterActiveCase);
+  const activeCasesByName = getCasesByName(activeCases);
```
[the remark this answers is not in the document]
Comment (TeaWithLucas) [reply app/server/controllers/active-cases.ts]: We can do

[the remark this answers is not in the document]
Comment (TeaWithLucas) [reply app/server/controllers/active-cases.ts]: Done

**Review by amoslv1 — COMMENTED** [review]
test/mock/s2s/services/oauth2_authoirse.ts typo in file name

**Review by amoslv1 — DISMISSED** [review]
test/mock/s2s/services/oauth2_authoirse.ts typo in file name

views/select-case.html does not have a corresponding njk file

**Comment by TeaWithLucas** [comment]
> test/mock/s2s/services/oauth2_authoirse.ts typo in file name
Fixed the name, I don't think it's even used
> views/select-case.html does not have a corresponding njk file
It does, its views/cases.njk

**Pull request merged**